### PR TITLE
Add DPR To Flatland Embedder

### PR DESCRIPTION
**Pull request by cfontas**
Have FlatlandExternalViewEmbedder handle DPR.

fxbug.dev/94000

Unittests in flatland_external_view_embedder_unittests and flatland_platform_view_unittests updated.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/fuchsia/flutter/flatland_platform_view.cc [hunk 1]
```diff
@@ -75,10 +75,14 @@ void FlatlandPlatformView::OnGetLayout(
...
-  // TODO(fxbug.dev/94000): Set device pixel ratio.
+  float pixel_ratio = 1.0f;
+  if (info.has_device_pixel_ratio()) {
+    view_pixel_ratio_ = info.device_pixel_ratio().x;
+    pixel_ratio = *view_pixel_ratio_;
+  }
 
   SetViewportMetrics({
-      1,                              // device_pixel_ratio
+      pixel_ratio,                    // device_pixel_ratio
```
Comment (uysalere) [inline shell/platform/fuchsia/flutter/flatland_platform_view.cc:85]: We can exercise this via FlatlandPlatformViewTests.SetViewportMetrics

File: shell/platform/fuchsia/flutter/flatland_external_view_embedder.cc [hunk 2]
```diff
@@ -172,6 +171,10 @@ void FlatlandExternalViewEmbedder::SubmitFrame(
   {
     TRACE_EVENT0("flutter", "SubmitLayers");
 
+    // First re-scale everything according to the DPR.
+    const float inv_dpr = 1.0f / frame_dpr_;
+    flatland_->flatland()->SetScale(root_transform_id_, {inv_dpr, inv_dpr});
+
```
[the remark this answers is not in the document]
Comment (cfontas) [reply shell/platform/fuchsia/flutter/flatland_external_view_embedder.cc:177]: SetImageDestinationSize just sets the scale on the internal leaf node for the image, so it'll also inherit the inv_dpr here

File: shell/platform/fuchsia/flutter/flatland_platform_view.cc (hunk 1 above)
Comment (cfontas) [reply shell/platform/fuchsia/flutter/flatland_platform_view.cc:87]: Done

File: shell/platform/fuchsia/flutter/flatland_platform_view.cc [hunk 3]
```diff
@@ -75,10 +75,14 @@ void FlatlandPlatformView::OnGetLayout(
   view_logical_size_ = {static_cast<float>(info.logical_size().width),
                         static_cast<float>(info.logical_size().height)};
 
-  // TODO(fxbug.dev/94000): Set device pixel ratio.
+  float pixel_ratio = 1.0f;
+  if (info.has_device_pixel_ratio()) {
+    view_pixel_ratio_ = info.device_pixel_ratio().x;
```
[the remark this answers is not in the document]
Comment (cfontas) [reply shell/platform/fuchsia/flutter/flatland_platform_view.cc:82]: Done

**Review by uysalere — APPROVED** [review]

**Pull request merged**